src/FSAL/FSAL_CORTXFS/handle.c

```diff
@@ -2833,37 +2864,52 @@ static fsal_status_t kvsfs_ftruncate(struct fsal_obj_handle *obj_hdl,
 				     struct state_t *state, bool bypass,
 				     struct stat *new_stat, int new_stat_flags)
 {
-	int rc;
+	int rc = 0;
 	fsal_status_t result = fsalstat(ERR_FSAL_NO_ERROR, 0);
 	struct kvsfs_file_state *fd = NULL;
 	struct kvsfs_fsal_obj_handle *obj = NULL;
 	cfs_cred_t cred = CFS_CRED_INIT_FROM_OP;
+	bool special_fd = false;
 
 	T_ENTER0;
 
 	assert(obj_hdl != NULL);
 	assert((new_stat_flags & STAT_SIZE_SET) != 0);
 	assert(obj_hdl->type == REGULAR_FILE);
 
-	/* Check if there is an open state which has O_WRITE openflag
-	 * set.
-	 */
-	result = kvsfs_find_fd(state, bypass, FSAL_O_WRITE, &fd);
-	if (FSAL_IS_ERROR(result)) {
-		goto out;
+	obj = container_of(obj_hdl, struct kvsfs_fsal_obj_handle, obj_handle);
+
+	if (state == NULL) {
+		/* special stateid handling required */
+		fd = gsh_calloc(1, sizeof(struct kvsfs_file_state));
+		result = kvsfs_get_special_fd(obj, FSAL_O_WRITE, fd);
```

Comment:
Just a doubt, is this fix needed to other callers of kvsfs_find_fd() where arg state can be NULL, i.e. kvsfs_read2, kvsfs_write2.

Reply:
Hi Pratyush, your understanding is correct. This fix would be needed for those functions as well.
Just holding it till we have clean test results for this change. Once everything looks good, we will have this fix for read & write paths as well.

Reply:
Thanks Sachin for explaining this.